scripts/llogcolor

```diff
@@ -95,71 +95,74 @@ def next_ansi_color():
     color_round_robin.append(color_round_robin.pop(0))  # rotate the list
     return next_color
 
-
-usage = """usage: %prog [options] [log1] [log2] ... [logN]
-
-%prog accepts a lustre log on stdin or as a filename on the command line,
-and outputs a colorized version of the log.  Lines are colored by task id.
-By default, and if stdout is a tty, %prog calls the pager (less) to display
-the colorized log.
-""".strip()
-
+description = (
+    "Accepts a lustre log on stdin or as a filename on the command line, "
+    "and outputs a colorized version of the log. "
+    "Lines are colored by task id. By default, and if stdout is a tty, "
+    "the pager (less) is called to display the colorized log."
+)
 
 def main():
     # Parse command-line options
-    p = optparse.OptionParser(usage=usage)
-    p.add_option(
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument(
         "-d",
         "--date",
         action="store_true",
         help="convert date to human-readable form",
     )
-    p.add_option(
+    parser.add_argument(
         "-s",
         "--split",
         action="store_true",
         help="create a separate file for each thread",
     )
-    p.add_option(
+    parser.add_argument(
         "-t",
         "--thread",
-        type="string",
+        type=str,
         metavar="TID",
         help="highlight one thread in a unique color (black on cyan)",
     )
-    p.add_option(
+    parser.add_argument(
         "-P",
         "--no-pager",
         action="store_false",
         dest="pager",
         help="disable the pager, just write to stdout",
     )
-    p.add_option(
+    parser.add_argument(
         "-C",
         "--no-color",
         action="store_false",
         dest="color",
         help="disable all colorization",
     )
-    p.set_defaults(color=True)
-    p.set_defaults(pager=True)
-    p.set_defaults(split=False)
-    p.set_defaults(date=False)
-    options, args = p.parse_args()
+    parser.add_argument(
+        "files",
+        nargs="*",
+        help="the log files to be processed",
+    )
+
+    args = parser.parse_args()
 
     if not sys.stdout.isatty():
-        options.pager = False
+        args.pager = False
 
     split_dir = "/tmp/llogcolor.%d" % (os.getpid())
     combined = os.path.join(split_dir, "combined")
 
-    if options.split:
+    if args.split:
         os.mkdir(split_dir)
         output = open(combined, "w")
         pager = None
-        pass
-    elif options.pager:
-        pager = subprocess.Popen(["less", "-KXRS"], stdin=subprocess.PIPE)
+    elif args.pager:
+        # open with encoding specified so that the subprocess
+        # can accept strings as input, not bytes
+        # this makes it behave like the other possible outputs
```

Comment:
In that it will accepts strings sent to it, and not bytes. In python2 they all accepted strings. However, by default in python3 the `subproccess` spawned will demand some kind of python byte object, and in python3 strings aren't byte objects anymore due to unicode.

Reply:
To be a little more detailed: There are 3 possible outputs.
The  console which is already set up and is `sys.stdout`, 
various temporary files, which are represented as python file objects,
and `less` which is a dealt with using `subprocess.run`.
To avoid having to check when it's time to write out data whether or not it should be in text mode or bytes mode, I changed the default behavior of the subprocess so that it matches the files and `sys.stdout`. Now all 3 require python string objects as input.

Reply:
I see.  How about making the comment "Specify an encoding so the subprocess can accept strings (not bytes).  Then our output to pager, files, and stdout, can all be the same (strings)."  and adding your description of the difference between python2 and python3 to the commit message?

Reply:
Sorry, I just realized you addressed this in the commit message already.

Reply:
I do have a mention of that in the commit message. Do you want it revised?